File: DSA/Array/TwoSum.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static void main(String[] args) {
        int[] nums = { 1, 7, 3, 2 };
        int target = 9;

        System.out.println(Arrays.toString(twoSum(nums, target)));
    }

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int remaining = target - nums[i];

            if (map.containsKey(remaining)) {
                return new int[] { map.get(remaining), i };
            }
            map.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: DSA/Array/Two_sum.py
def two_sum(arr, target):
    nums_dict = {}
    for i, j in enumerate(arr):
        remaining = target - j

        if remaining in nums_dict:
            return [nums_dict[remaining], i]
        
        nums_dict[j] = i
        
arr = [1, 7, 3, 2]
target = 9
print(two_sum(arr, target))
